plugin-ci/orb.yml

```diff
@@ -104,11 +104,21 @@ jobs:
       - checkout
       - *restore_cache
       - run: make dist
+      - run:
+          name: "Generating release notes"
+          command: |
+            printf "Supported Mattermost Server Versions: **$(cat plugin.json | jq .min_server_version -r)+** \n## Enhancements\n\n## Fixes\n"  >> dist/release-notes.md
+            if [[ $(git tag -l | wc -l) -eq 1 ]]; then
+              git log --pretty='format:- %h %s' --abbrev-commit --no-decorate --no-color $(git rev-list --max-parents=0 HEAD) HEAD >> dist/release-notes.md
```

Comment:
@hanzei why not using something like [git-chglog](https://github.com/git-chglog/git-chglog). It's way more flexible and cleaner

Reply:
As for the point @jfrerich mentioned this tool (as well as others) looks for some pattern in the commit message. something like `<type>(<area>): <actual commit message>`, in which:

Reply:
Thanks you very much for sharing your ideas @khos2ow :+1: 

First of all, I agree that https://github.com/git-chglog/git-chglog is an awesome tool for generating and maintaining change logs. I quite like to go templating part of it.

> and one final note that unfortunately you cannot start using the convention in the middle of a release, because you'd lose half of the history before start using the convention, the cut off for using this new approach should be on starting a new release.

That is one point that makes this hard to use for **all** mattermost-plugins. I want something that just works, as it's very hard to enforce a style for all plugin repositories (not only the ones inside the Mattermost organisation).

I could see an **opt in** version of this orb that uses `git-chglog`.



> As for an example everything in this [release note](https://github.com/segmentio/terraform-docs/releases/tag/v0.8.0) except 1) one bold line at the top and 2) list of `Contributors` was automatically created using the git-chglog tool and custom configuration.

Could you please point out how you generated the `Contributors` list?

Reply:
> That is one point that makes this hard to use for all mattermost-plugins. I want something that just works, as it's very hard to enforce a style for all plugin repositories (not only the ones inside the Mattermost organisation).

This is basically what can be achieved by using `git-chglog`, one configuration per mattermost organization which all the plugins can use (basically they need to copy over to their repositories), and they continue as nothing has changed, the difference is **if** they want to opt-in with the newer format of release note, they have to hold on to it until the start of the new release. If or when they want to opt-in, they only need to start using `<type>: ` prefix in commit messages, otherwise nothing would change and they continue without extra prefix in commit messages.

Going to back [this example](https://github.com/segmentio/terraform-docs/blob/master/CHANGELOG.md#v070---2019-12-12) I've added the support for `git-chglog` sometime in `v0.7.0` but didn't actually start using the `<type>: <message>` convention until start of `v0.8.0`.

Also there's something I didn't quite understand, there is this 
```
... ## Enhancements\n\n## Fixes\n"  >> dist/release-notes.md
```
How do you put log lines under each section? because in the following next lines there's
```
git log  .... >> dist/release-notes.md
```

Because of low volume of contributors I actually haven't attempted to automate it, I execute this Makefile [target](https://github.com/segmentio/terraform-docs/blob/master/Makefile#L142) (we need to commit this anyway) and based on the diff I either remember their handle or look them up. But [this](https://github.com/jonschlinkert/github-contributors) seems to be a good starting point (not sure how configurable it is though).

Reply:
> This is basically what can be achieved by using `git-chglog`, one configuration per mattermost organization which all the plugins can use (basically they need to copy over to their repositories)

This is the issue. By coping the config file over to the repositories, central becomes very hard. Do you see a way to manage this in a central place easily?

Reply:
That's a valid point, `git-chglog` essentially only needs a `config.yml` file and a `template.md` file, so what I can think of at the moment is to _wget_ them from a _somewhere_ (I don't know where exactly though, may be a common mattermost config repo exists? a folder within this repo itself?).
There's another idea that the orb wraps all the required files inside it (I have to check if this is possible at all or not and what happens to relative path of the files when orb is run on repos, but this could be interesting if it was possible).